syntax = "proto3";
package task;




import "capacity_admin_sdk/model/capacity_admin/portrait.proto";

/*
GetTaskDetail请求
*/
message GetTaskDetailRequest {
    /*
    任务 Id
    */
    string taskId = 1 ;
}

/*
GetTaskDetail返回
*/
message GetTaskDetailResponse {
    /*
    资源画像列表
    */
    repeated capacity_admin.Portrait portrait = 1 ;
    /*
    任务ID
    */
    string instanceId = 2 ;
    /*
    任务名称
    */
    string name = 3 ;
    /*
    主机列表
    */
    repeated string hosts = 4 ;
    /*
    疑似闲置主机列表
    */
    repeated string suspectedIdleHosts = 5 ;
    /*
    闲置主机列表
    */
    repeated string idleHosts = 6 ;
    /*
    非闲置主机列表
    */
    repeated string nonIdleHosts = 7 ;
    /*
    任务状态, executing, confirmation, confirmed, failed
    */
    string status = 8 ;
    /*
    任务分析完成时间
    */
    string endTime = 9 ;
    /*
    命中率
    */
    float hitRate = 10 ;
    /*
    任务创建时间
    */
    string ctime = 11 ;
    /*
    任务创建者
    */
    string creator = 12 ;
    /*
    工具执行 Id
    */
    string execId = 13 ;
    /*
    分析正常主机列表
    */
    repeated string normalHosts = 14 ;
    /*
    分析异常主机列表
    */
    repeated string abnormalHosts = 15 ;
}

/*
GetTaskDetailApi返回
*/
message GetTaskDetailResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    task.GetTaskDetailResponse data = 4 ;
}

